files/hipchat.rb

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env ruby
+
+require "#{File.dirname(__FILE__)}/base"
+#require_relative 'base'
+
+class Hipchat < BaseHandler
+  def handle
+    timeout_and_retry do
+      case @event['check']['status'].to_i
+      when 1,2
+        trigger_incident
+      when 0
+        resolve_incident
+      end
+    end
+  end
+
+  def api_key
+    handler_settings['api_key'] || false
+  end
+
+  def trigger_incident
+    return false unless api_key
+    alert_hipchat(
+      hipchat_room,
+      'sensu',
+      hipchat_message,
+      { :color => hipchat_message_colour, :notify =>  true }
+    )
+  end
+
+  def resolve_incident
+    return false unless api_key
+    alert_hipchat(
+      hipchat_room,
+      'sensu',
+      hipchat_message,
+      { :color => hipchat_message_colour }
+    )
+  end
+
+  def event_time
+    Time.at(@event['check']['issued']).utc.strftime "%Y-%m-%d %H:%M:%S UTC"
+  end
+
+  def hipchat_message
+"
+<b>#{event_time} - #{@event['check']['name']} on #{@event['client']['name']} (#{@event['client']['address']}) - #{human_check_status}</b>
+<br /><br />
+&nbsp;&nbsp;#{@event['check']['notification'] || @event['check']['output']}
+"
+  end
+
+  def hipchat_message_colour
+    case @event['check']['status']
+    when 0
+      'green'
+    when 1
+      'yellow'
+    when 2
+      'red'
+    else
+      'grey'
+    end
+  end
+
+  def default_room
+    handler_settings['hipchat_room']
+  end
+
+  def hipchat_room
+    team_data('hipchat_room') || default_room || nil
```

Comment:
I'm not as concerned about the team hash. Yes, I think the default room should be set on a per-team basis.
I was more talking about extracting a room to use from the event data. You are not doing that yet, but every other handler has that flexibility.

I just don't want this to make the same mistake I made and "encode" the idea that it is hipchat-specific. The team data can have hipchat stuff, but the *client* interface (sending data to the local socket, sensu checks in puppet) should not be "aware" that things are getting routed to hipchat or irc (or pagerduty or opsgenie)

```
sensu_handlers::teams:
  ops:
    notication_email: ops@domain
    hipchat_room: foo
```

This is the first part, but like all the other handlers, it should respond to self-describing event data:

```
echo '{
  "team": "ops",
  "channels": ["ops-notifications"],
  "page": false,
  "ticket: true,
  "output": "something is wrong",
  "status": 2
}' > /dev/tcp/localhost/3030
```